packages/remix-react/single-fetch.tsx

```diff
@@ -24,6 +29,23 @@ import { escapeHtml } from "./markup";
 import type { RouteModules } from "./routeModules";
 import invariant from "./invariant";
 
+// clientLoader
+type ClientLoaderArgs = RRLoaderArgs<undefined> & {
+  serverLoader: <T extends unstable_Loader>() => Promise<unstable_Serialize<T>>;
+};
+export type ClientLoader = (args: ClientLoaderArgs) => unknown;
+export let defineClientLoader = <T extends ClientLoader>(
+  clientLoader: T
+): T & { hydrate?: boolean } => clientLoader;
+
+// clientAction
+type ClientActionArgs = RRActionArgs<undefined> & {
+  serverAction: <T extends unstable_Action>() => Promise<unstable_Serialize<T>>;
+};
+export type ClientAction = (args: ClientActionArgs) => unknown;
+export let defineClientAction = <T extends ClientAction>(clientAction: T): T =>
+  clientAction;
```

Comment:
These need to be exported from the react package to avoid pulling `@remix-run/node` stuff into client bundles